libraries/chain/transaction_context.cpp

```diff
@@ -459,7 +459,10 @@ namespace eosio::chain {
          return;
 
       auto now = fc::time_point::now();
-      if( explicit_billed_cpu_time || deadline_exception_code == deadline_exception::code_value ) {
+      if (explicit_billed_cpu_time && block_deadline > now) {
+         EOS_THROW( interrupt_exception, "interrupt signaled, ran ${bt}us, start ${s}",
+                    ("bt", now - pseudo_start)("s", start) );
+      } else if( explicit_billed_cpu_time || deadline_exception_code == deadline_exception::code_value ) {
          EOS_THROW( deadline_exception, "deadline exceeded ${billing_timer}us",
                      ("billing_timer", now - pseudo_start)("now", now)("deadline", _deadline)("start", start) );
```

Comment:
```suggestion
                     ("billing_timer", now - pseudo_start) );
```

Reply:
Why? That data is still in the exception if you print out the details.

Reply:
I didn't see (and still don't) where "now", "deadline" are "start" are referenced, so I thought they were unused.

Reply:
They are not referenced in the string. However, non-referenced captured values are printed when details of exceptions are requested.

See https://github.com/AntelopeIO/spring/blob/bb4d159218eb5fb6603f0c82702800c7bd4dae90/libraries/libfc/src/exception.cpp#L174-L174

Reply:
Oh, cool, I didn't know that!